### ISSUE TITLE
Error in barshadow

### ISSUE THREAD

**Issue by hbushouse** [spacetelescope/jwst#2314]
John Scott reports the following calspec2 error during I&T testing of B7.1.3:
```
We are seeing 3 spec2 failures in program 95065 for the odd reason of
converting NaN to an integer.

There errors all look like this:

ERROR RUNNING STEP 'Spec2Pipeline':
cannot convert float NaN to integer
Traceback (most recent call last):
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/stpipe/cmdline.py",
line 297, in step_from_cmdline
step.run(*positional)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/stpipe/step.py",
line 398, in run
step_result = self.process(*args)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py",
line 86, in process
asn.filename
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py",
line 227, in process_exposure_product
input = self.barshadow(input)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/stpipe/step.py",
line 398, in run
step_result = self.process(*args)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/barshadow/barshadow_step.py",
line 47, in process
result = bar_shadow.do_correction(input_model, barshadow_model)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/barshadow/bar_shadow.py",
line 84, in do_correction
correction = interpolate(yrow, wcol, shadow)
File
"/ifs/int/jwstb/pipeline/pkgs/miniconda3/envs/jwstdp-0.9.6-py35/lib/python3.5/site-packages/jwst-0.9.6-py3.5-linux-x86_64.egg/jwst/barshadow/bar_shadow.py",
line 417, in interpolate
ix = int(array_column)
ValueError: cannot convert float NaN to integer

See a log file example here:
/ifs/int/jwstb/info/owl/logs/jwowlmgr_jw95065-o008_20180724t211055_spec2_002_1532472313.056290
```

**Comment by stscirij** [comment]
I ran the spec2 pipeline on exactly the same data file (jw95065008001_02101_00001_nrs1_rate.fits) with exactly the same reference files and ancilliary files and the pipeline ran to completion, including, of course, the barshadow step.  I checked the text output against that in the SDP log and, apart from some debug output from CRDS, the logs were identical up to the error in the barshadow step.

**Comment by hbushouse** [comment]
John was running the B7.1.3 (v0.9.6) release. Did you test it against that version or the current dev?

**Comment by stscirij** [comment]
I tested against current, now testing against 0.9.6.

**Comment by stscirij** [comment]
Now I see the same error in the barshadow step

**Comment by stscirij** [comment]
Fixed by #2384
